src/runtime.ts

```diff
@@ -0,0 +1,305 @@
+import type { Redis } from "ioredis";
+import { _Private, Publisher, Subscriber } from "./types";
+import { CreateResumableStreamContextOptions } from "./types";
+import { ResumableStreamContext } from "./types";
+
+interface CreateResumableStreamContext {
+  keyPrefix: string;
+  waitUntil: (promise: Promise<unknown>) => void;
+  subscriber: Subscriber;
+  publisher: Publisher;
+}
+
+export function createResumableStreamContextFactory(defaults: _Private.RedisDefaults) {
```

Comment:
The main refactor is here - changing this into a factory which can take in configurable defaults.